scheduler/local_scd_server.py

```diff
@@ -0,0 +1,320 @@
+
+SWG_GIT_REPO_DIR = 'schedules'
+SWG_GIT_REPO = "https://github.com/SuperDARN/schedules.git"
+
+EXPERIMENTS = {
+    "sas" : {
+              "common_time" : "twofsound",
+              "discretionary_time" : "twofsound",
+              "htr_common_time" : "twofsound",
+              "themis_time" : "themisscan",
+              "special_time_normal" : "twofsound",
+              "rbsp_time" : "rbspscan",
+              "no_switching_time" : "normalscan",
+              "interleaved_time" : "interleavedscan"
+    },
+    "pgr" : {
+              "common_time" : "twofsound",
+              "discretionary_time" : "twofsound",
+              "htr_common_time" : " twofsound",
+              "themis_time" : "themisscan",
+              "special_time_normal" : "normalscan",
+              "rbsp_time" : "rbspscan",
+              "no_switching_time" : "normalscan",
+              "interleaved_time" : "interleavedscan"
+    },
+    "rkn" : {
+              "common_time" : "twofsound",
+              "discretionary_time" : "twofsound",
+              "htr_common_time" : "twofsound",
+              "themis_time" : "themisscan",
+              "special_time_normal" : "twofsound",
+              "rbsp_time" : "rbspscan",
+              "no_switching_time" : "normalscan",
+              "interleaved_time" : "interleavedscan"
+    },
+    "inv" : {
+              "common_time" : "twofsound",
+              "discretionary_time" : "twofsound",
+              "htr_common_time" : "twofsound",
+              "themis_time" : "themisscan",
+              "special_time_normal" : "twofsound",
+              "rbsp_time" : "rbspscan",
+              "no_switching_time" : "normalscan",
+              "interleaved_time" : "interleavedscan"
+    },
+    "cly" : {
+              "common_time" : "twofsound",
+              "discretionary_time" : "twofsound",
+              "htr_common_time" : "twofsound",
+              "themis_time" : "themisscan",
+              "special_time_normal" : "twofsound",
+              "rbsp_time" : "rbspscan",
+              "no_switching_time" : "normalscan",
+              "interleaved_time" : "interleavedscan"
+    }
+}
+class SWG(object):
+    """Holds the data needed for processing a SWG file.
+
+    Attributes:
+        scd_dir (TYPE): Description
+    """
+    def __init__(self, scd_dir):
+        super(SWG, self).__init__()
+        self.scd_dir = scd_dir
+
+        try:
+            cmd = "git -C {}/{} rev-parse".format(self.scd_dir, SWG_GIT_REPO_DIR)
+            sp.check_output(cmd, shell=True)
+        except sp.CalledProcessError as e:
+            cmd = 'cd {}; git clone {}'.format(self.scd_dir, SWG_GIT_REPO)
+            sp.call(cmd, shell=True)
+
+
+    def new_swg_file_available(self):
+        """Checks if a new swg file is uploaded via git.
+
+        Returns:
+            TYPE: True, if new git update is available.
+        """
+        cmd = "cd {}/{}; git log ..origin/master --oneline | wc -l".format(self.scd_dir,
```

Comment:
can you give a comment about how this works? Wouldn't it always return true if there are any commits in the log?

Reply:
It basically returns to you the number of new commits in a given branch.

Reply:
I will comment on this.

Reply:
done.